netlify/edge-functions/workspaces-card.tsx

```diff
@@ -0,0 +1,282 @@
+import React from "react";
+import { ImageResponse } from "og_edge";
+import type { Config } from "https://edge.netlify.com";
+
+const baseApiUrl = Deno.env.get("NEXT_PUBLIC_API_URL");
+
+// TODO: truncate name and description
+
+function getLocalAsset(url: URL): Promise<ArrayBuffer> {
+  return fetch(url).then((res) => res.arrayBuffer());
+}
+
+function getOrgUsernameAvatar(username: string, size = 25.2) {
+  return `https://www.github.com/${username}.png?size=${size}`;
+}
+
+const getActivityRatio = (total?: number) => {
+  if (total === undefined) {
+    return "-";
+  }
+
+  if (total > 7) {
+    return "high";
+  }
+
+  if (total >= 4 && total <= 7) {
+    return "medium";
+  }
+
+  return "low";
+};
+
+export default async function handler(req: Request) {
+  const { searchParams, pathname } = new URL(req.url);
+  const workspaceName = searchParams.get("wname");
+  const workspaceDescription = searchParams.get("description");
+  const [workspaceId, range = 30] = pathname.split("/").slice(-2);
+
+  const [repoStatsResponse, workspaceReposResponse] = await Promise.all([
+    fetch(new URL(`${baseApiUrl}/workspaces/${workspaceId}/stats?range=${range}`, baseApiUrl)),
+    fetch(new URL(`${baseApiUrl}/workspaces/${workspaceId}/repos?range=${range}&limit=3`, baseApiUrl)),
+  ]);
+
+  const repoStats = (await repoStatsResponse.json()) as Record<string, Record<string, number>>;
+  const workspaceRepos = (await workspaceReposResponse.json()) as Record<string, string>[];
+  const restOfReposCount = workspaceRepos.meta.itemCount < 4 ? 0 : workspaceRepos.meta.itemCount - 3;
+  const workspaceReposNames = workspaceRepos.data.slice(0, 3).map(({ repo }) => repo.full_name);
+
+  if (!workspaceName) {
+    return new Response("A workspace name must be specified", { status: 404 });
+  }
+
+  const logoImg = getLocalAsset(new URL("/assets/og-images/workspaces/open-sauced-logo.png", req.url));
+  const interSemiBoldFont = getLocalAsset(new URL("/assets/card/Inter-SemiBold.ttf", req.url));
+  const interBlackFont = getLocalAsset(new URL("/assets/card/Inter-Black.ttf", req.url));
+
+  const [interSemiBoldFontData, interBlackFontData, logoImgData] = await Promise.all([
+    interSemiBoldFont,
+    interBlackFont,
+    logoImg,
+  ]);
+
+  const statContainerStyles = {
+    display: "flex",
+    alignItems: "baseline",
+    gap: "10px",
+  };
+
+  const statIconStyles = {
+    width: 32,
+    height: 32,
+  };
+
+  const statTextContainerStyles = {
+    position: "relative",
+    top: "2px",
+    display: "flex",
+    alignItems: "baseline",
+    gap: "12px",
+  };
+
+  const statValueContainerStyles = {
+    overflow: "hidden",
+    position: "relative",
+    maxHeight: "42px",
+  };
+
+  const statValueStyles = {
+    position: "relative",
+    top: "2px",
+    fontSize: "42px",
+    fontWeight: 700,
+  };
+
+  const statSubTextStyles = {
+    position: "relative",
+    top: "3px",
+    fontSize: "24px",
+    fontWeight: 500,
+  };
+
+  return new ImageResponse(
```

Comment:
I tried breaking some of this up into components, but it doesn't appear to work. I'm not positive, but this might be a limitation of the `ImageResponse`.